### PR TITLE
Add metadata relocation documentation

### PR DESCRIPTION
**Pull request by amzn-mike**
## Change summary

Added documentation for metadata asset relocation system.

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/assets/pipeline/metadata.md
Comment (AMZN-stankowi) [inline content/docs/user-guide/assets/pipeline/metadata.md:1]: Are you planning on making another page on guiding people through the process in more detail? Understanding the gaps that aren't handled (path based references, especially from code) is important. Otherwise, that could be a section at the bottom of this document.

Maybe we should also make sure github issues are cut for each asset type with known gotchas around path based references, and link those here, to track the status for primary asset types supported in the engine by default.

Comment (amzn-mike) [reply content/docs/user-guide/assets/pipeline/metadata.md:1]: I added a limitations section to talk about this.  Does that feel sufficient?

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+* If blue.png is renamed to red.png, the metadata file blue.png.meta must be renamed to red.png.meta.
+* If textures/hat.png is moved to character/hat.png, the metadata file must be moved to character/hat.png.meta.
+
+
+If using source control, metadata files **must** be checked in along with the source asset and updated with any move/rename changes.
+
+There is one exception to the above, which is Intermediate Asset metadata files.  These metadata files follow the same rules as Intermediate Assets: they should not be modified or checked into source control.
+
+## How to Enable
+To enable a file type, create a .setreg file with the following settings:
```
[the remark this answers is not in the document]
Comment (amzn-mike) [reply content/docs/user-guide/assets/pipeline/metadata.md:31]: Good suggestion, added

File: content/docs/user-guide/assets/pipeline/metadata.md
Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/pipeline/metadata.md:1]: Yeah. You might be able to word smith it more to not sound like it doesn't support renaming and more "If you rename a file and there's an incoming path reference, you'll need to update that path". So people can still use this solution with those files because some cases are handled, but not every case.

**Review by AMZN-stankowi — APPROVED** [review]

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+---
+linkTitle: Metadata Asset Relocation
+title: Metadata Asset Relocation
+description: In Open 3D Engine (O3DE), Metadata Asset Relocation allows files to be moved and renamed freely without breaking existing references by storing a UUID in a side-car file (.meta).
+weight: 600
+toc: true
+---
+
+Metadata Asset Relocation allows files to be moved and renamed freely without breaking existing references by storing a UUID in a side-car file (.meta).  See here for technical details on how the system works: https://github.com/o3de/sig-content/blob/main/rfcs/rfc-77-metadata-asset-relocation.md
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:9]:
```suggestion
Metadata Asset Relocation allows files to be moved and renamed freely without breaking existing references by storing a UUID in a side-car file (.meta).  See the [Asset Metadata Relocation RFC](https://github.com/o3de/sig-content/blob/main/rfcs/rfc-77-metadata-asset-relocation.md) for technical details on how the system works: 

```

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+linkTitle: Metadata Asset Relocation
+title: Metadata Asset Relocation
+description: In Open 3D Engine (O3DE), Metadata Asset Relocation allows files to be moved and renamed freely without breaking existing references by storing a UUID in a side-car file (.meta).
+weight: 600
+toc: true
+---
+
+Metadata Asset Relocation allows files to be moved and renamed freely without breaking existing references by storing a UUID in a side-car file (.meta).  See here for technical details on how the system works: https://github.com/o3de/sig-content/blob/main/rfcs/rfc-77-metadata-asset-relocation.md
+
+By default, the system is currently disabled as not all file types are currently able to be relocated without issue.  File types must be enabled individually.  Once a type is enabled, the Asset Processor will generate metadata files for every file of the given type on startup.  Any file which has a metadata associated with it can then be moved/renamed.  The Asset Processor does not need to be running when moving/renaming files and there are no special tools required to do so.
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:11]:
```suggestion
By default, the metadata system is currently disabled because not all file types can be relocated without issue currently.  Each asset file types must be enabled individually in settings.  Once an asset file type is enabled, the Asset Processor will generate a metadata file for every file of the given type on startup.  Any file which has a metadata associated with it can then be moved or renamed.  The Asset Processor does not need to be running when moving or renaming files and there are no special tools required to do so.

```

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+## How to use
+Moving/renaming assets can be done either in the editor via the asset browser or using any file management tool of your choosing.
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:14]:
```suggestion
Moving or renaming assets can be done either in the editor via the asset browser or using any file management tool of your choosing, such as the File Explorer in Windows.

```

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+There is only one requirement when **moving/renaming an asset: the metadata file must be moved/renamed as well**.
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:16]:
```suggestion
There is only one requirement when moving or renaming an asset, which is **the metadata file must be moved or renamed as well**.

```

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+Folders can be moved/renamed without having to touch metadata files, but note that all files within a folder must have a metadata file to ensure existing references do not break.
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:18]:
```suggestion
Folders can be moved or renamed without having to touch metadata files, but that all files within a folder **must have a metadata file** to ensure existing references do not break.

```

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+Examples:
+
+* If blue.png is renamed to red.png, the metadata file blue.png.meta must be renamed to red.png.meta.
+* If textures/hat.png is moved to character/hat.png, the metadata file must be moved to character/hat.png.meta.
+
+
+If using source control, metadata files **must** be checked in along with the source asset and updated with any move/rename changes.
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:26]:
```suggestion
If using source control, metadata files **must be checked in** along with the source asset and updated to match any move or rename changes to those assets.

```

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+                    ]
+                }
+            }
+        }
+    }
+}
+```
+EnabledTypes is simply an array of strings of file extension to enable.  For example, `".txt", ".png"`
+
+It is highly recommened that this setreg file be placed in a central location which is shared with other users of the project to ensure everyone is generating the same metadata files.  For a game project, it should be placed in the project directory; for gem development, it should be placed in the gem directory.  Keep in mind this is an Asset Processor wide setting, it will apply to all files the AP processes, not just the ones in the project/gem where the setting exists.
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:52]:
```suggestion
It is highly recommended that this setreg file be placed in a central location, which is shared with other users of the project, to ensure everyone is generating the same metadata files.  For a game project, it should be placed in the project directory; for gem development, it should be placed in the gem directory.  Keep in mind this is an Asset Processor (AP) wide setting, it will apply to all files the AP processes, not just the ones in the project or gem where the setting exists.

```

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+Once a metadata file is generated for an asset and the asset is relocated or any references are saved using the newly generated UUID, the metadata file must be kept alongside the asset.  Not doing so could break references to the file.
+
+Note that AP will continue to use existing metadata files for assets which have them regardless of the above settings.
+
+## Renaming while AP is running
+By default, renaming assets manually while Asset Processor is running can prove difficult as AP will try to immediately create a new metadata file for the renamed asset.  To prevent this, AP can be configured to wait for a specified time before it attempts to process a "new" asset, which avoids the automatic creation of an unwanted, new metadata file.  Note that this delay will affect all newly created assets for metadata-enabled types, which is why this is off by default.  The longer the delay time, the longer it will take AP to start processing any newly created metadata-enabled file type.  In a .setreg file, add the following setting:
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:59]:
```suggestion
By default, renaming assets manually while Asset Processor is running can prove difficult as AP will try to immediately create a new metadata file for the renamed asset.  To prevent this, AP can be configured to wait for a specified time before it attempts to process a "new" asset, which avoids the automatic creation of an unwanted, new metadata file.  

> This delay will affect all newly created assets for metadata-enabled types and is off by default.  The longer the delay time, the longer it will take AP to start processing any newly created metadata-enabled file type.

In a .setreg file, add the following setting:

```

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+                "Metadata":
+                {
+                    "MetaCreationDelayMs": 5000
+                }
+            }
+        }
+    }
+}
+```
+The value specified is the milliseconds to wait.  The above example config will wait 5 seconds before processing an asset.  Note that this setting can be user-specific and does not need to be shared across users if not desired.
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:77]:
```suggestion
The value specified is the milliseconds to wait.  The above example config will wait 5 seconds before processing an asset.  

>  This setting can be user-specific and does not need to be shared across users if not desired.

```

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+This setting is mainly useful to help allow manual renaming of assets, but can help in some situations of slow file updates on disk, such as copying a large number of files or fetching latest from source control.  These operations can sometimes be ordered in a way that requires a very high delay setting, such as a Git LFS fetch which may fetch files in groups by type, possibly resulting in metadata files being created on disk long after the source asset was created.  In these situations, it is recommended to close AP first to avoid conflicts caused by AP creating metadata files which already exist in source control.
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:79]:
```suggestion
This setting is mainly useful to help allow manual renaming of assets, but can help in some situations of slow file updates on disk, such as copying a large number of files or fetching latest from source control.  These operations can sometimes be ordered in a way that requires a very high delay setting, such as a Git LFS fetch, which may fetch files in groups by type, possibly resulting in metadata files being created on disk long after the source asset was created.  In these situations, it is recommended to close AP first to avoid conflicts caused by AP creating metadata files which already exist in source control.

```

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+
+## Asset Processor Batch
+Since Asset Processor Batch is intended for use with automated processes, it will not create metadata files and will not use the meta creation delay, regardless of what settings are configured.  It will make use of any existing metadata files however.
```
[the remark this answers is not in the document]
Comment (amzn-mike) [reply content/docs/user-guide/assets/pipeline/metadata.md:97]: AP batch is mostly used for Jenkins and other automated processes, so we usually don't want Jenkins to be creating metadata files

[the remark this answers is not in the document]
Comment (AMZN-stankowi) [reply content/docs/user-guide/assets/pipeline/metadata.md:90]: Not having Asset Processor Batch create meta files by default is a good call and good idea. I think this as the default behavior makes sense. It's probably worth eventually having a flag to enable meta file generation, but not as necessary. That would allow for teams to handle edge cases themselves, and more options for building automated tests.

**Review by AMZN-stankowi — APPROVED** [review]

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
+---
+linkTitle: Metadata Asset Relocation
+title: Metadata Asset Relocation
+description: In Open 3D Engine (O3DE), Metadata Asset Relocation allows files to be moved and renamed freely without breaking existing references by storing a UUID in a side-car file (.meta).
+weight: 600
+toc: true
+---
+
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:8]: I would love to see something like:

```
{{< note >}}
This feature is a work-in-progress. Don't depend on it yet; it may go through significant changes.
{{< /note >}}

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+```
+EnabledTypes is simply an array of strings of file extension to enable.  For example, `".txt", ".png"`
+
+It is highly recommended that this setreg file be placed in a central location, which is shared with other users of the project, to ensure everyone is generating the same metadata files.  For a game project, it should be placed in the project directory; for gem development, it should be placed in the gem directory.  Keep in mind this is an Asset Processor (AP) wide setting, it will apply to all files the AP processes, not just the ones in the project or gem where the setting exists.
+
+Once a metadata file is generated for an asset and the asset is relocated or any references are saved using the newly generated UUID, the metadata file must be kept alongside the asset.  Not doing so could break references to the file.
+
+> AssetProcessor will continue to use existing metadata files for assets which have them, regardless of the above settings.
+
+## Renaming while AP is running
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:59]:
```suggestion
## Renaming while Asset Processor is running

```

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+The value specified is the milliseconds to wait.  The above example config will wait 5 seconds before processing an asset.  
+
+>  This setting can be user-specific and does not need to be shared across users if not desired.
+
+This setting is mainly useful to help allow manual renaming of assets, but can help in some situations of slow file updates on disk, such as copying a large number of files or fetching latest from source control.  These operations can sometimes be ordered in a way that requires a very high delay setting, such as a Git LFS fetch, which may fetch files in groups by type, possibly resulting in metadata files being created on disk long after the source asset was created.  In these situations, it is recommended to close AP first to avoid conflicts caused by AP creating metadata files which already exist in source control.
+
+## Asset Processor Batch
+Since Asset Processor Batch is intended for use with automated processes, it will not create metadata files and will not use the meta creation delay, regardless of what settings are configured.  It will make use of any existing metadata files however.
+
+## Limitations
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:92]: To learn more of the technical contribution/setup perhaps add a reference to the engine contributor doc here? Or cover in limitations ie. While feature is still experimental, see the engine contributor docs for further information?

Think it would be nice to highlight how others know more.

**Review by lmbr-pip — COMMENTED** [review]
This needs to be linked from somewhere else otherwise its will not be discoverable. If we do that please use the  (EXPERIMENTAL)  tag - Metadata Asset Relocation  (EXPERIMENTAL) - <link to this page>

**Comment by lmbr-pip** [comment]
SIG/docs maintainers please note this is a **Development** only change and should not be merged for the 2305 release (please do not merge to stabilization/2305 or main at this time)

File: content/docs/user-guide/assets/pipeline/metadata.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
+---
+linkTitle: Metadata Asset Relocation
+title: Metadata Asset Relocation
+description: In Open 3D Engine (O3DE), Metadata Asset Relocation allows files to be moved and renamed freely without breaking existing references by storing a UUID in a side-car file (.meta).
+weight: 600
+toc: true
+---
+
+Metadata Asset Relocation allows files to be moved and renamed freely without breaking existing references by storing a UUID in a side-car file (.meta).  See the [Asset Metadata Relocation RFC](https://github.com/o3de/sig-content/blob/main/rfcs/rfc-77-metadata-asset-relocation.md) for technical details on how the system works
```
Comment (lmbr-pip) [inline content/docs/user-guide/assets/pipeline/metadata.md:9]:
```suggestion
Metadata Asset Relocation is an experimental feature that allows files to be moved and renamed freely without breaking existing references by storing a UUID in a side-car file (.meta).  See the [Asset Metadata Relocation RFC](https://github.com/o3de/sig-content/blob/main/rfcs/rfc-77-metadata-asset-relocation.md) for technical details on how the system works

```

**Comment by chanmosq** [comment]
> SIG/docs maintainers please note this is a **Development** only change and should not be merged for the 2305 release (please do not merge to stabilization/2305 or main at this time)

Noted. I've added a label indicating that this must remain in development only. We will avoid merging this to stabilization.

File: content/docs/user-guide/assets/pipeline/metadata.md (hunk 15 above)
Comment (amzn-mike) [reply content/docs/user-guide/assets/pipeline/metadata.md:94]: Added a link to the contributors doc

File: content/docs/user-guide/assets/pipeline/metadata.md (hunk 13 above)
Comment (amzn-mike) [reply content/docs/user-guide/assets/pipeline/metadata.md:8]: I applied the rewording to the first paragraph to mention this is experimental.  Should this note still be added as well?

**Comment by chanmosq** [comment]
@lmbr-pip @amzn-mike  Is it possible to hold off on merging this until after the release on May 3? That will make merging dev to stabilization easier, and avoid issues with cherry-picking.

**Comment by lmbr-pip** [comment]
> @lmbr-pip @amzn-mike Can we hold off on merging this until after the release on May 3? That will make merging dev to stabilization easier, and avoid issues with cherry-picking.

That should be ok, it does make collaborating on future updates harder, but we can just add any updates to this PR if needed.

**Review by micronAMZN — APPROVED** [review]
